### PR TITLE
fix(avatars): Add type to Fetch avatar

### PR DESCRIPTION
**Pull request by shahzadaziz**
Description : 

The Unified Shared modal component receive callback functions to fetch Avatar URLs.
This fix add contact `type` to the callback function. The contact type is already available in in the Contact object ( 'user' or 'group') and is returned as part of API response. 

The 'type' param and prop is set as optional, and can be null or undefined if it's not present in the contact object.

### CODE REVIEW

**Comment by mickr** [comment]
Can you please add a description for the PR?

File: src/components/contact-datalist-item/ContactDatalistItem.tsx [hunk 1]
```diff
@@ -15,6 +15,7 @@ export interface ContactDatalistItemProps {
     name: string | null | undefined;
     showAvatar?: boolean;
     subtitle?: React.ReactNode;
+    type?: string;
```
Comment (bfoxx1906) [inline src/components/contact-datalist-item/ContactDatalistItem.tsx:18]: Can type ever be null or undefined? If so we might need to treat this similar to how we do the name e.g. type?: string|null|undefined

**Review by greg-in-a-box — CHANGES_REQUESTED** [review]
Can we add some more unit tests, the snapshot that was updated is only covering the test coverage from feature level but not at the component level where they could be used in other places outside of USM

**Comment by shahzadaziz** [comment]
@greg-in-a-box added some tests. Can you take a look.

**Review by greg-in-a-box — APPROVED** [review]

**Review by JChan106 — APPROVED** [review]

**Pull request merged**